src/ext/handlersFilter.c

```diff
@@ -0,0 +1,357 @@
+}
+
+static RdbRes filterHashPlain(RdbParser *p, void *userData, RdbBulk field, RdbBulk value) {
+    UNUSED(p, field, value);
+    return ((RdbxFilter *) userData)->cbReturnValue;
+}
+
+static RdbRes filterSetMember(RdbParser *p, void *userData, RdbBulk member) {
+    UNUSED(p, member);
+    return ((RdbxFilter *) userData)->cbReturnValue;
+}
+
+static RdbRes filterSetPlain(RdbParser *p, void *userData, RdbBulk item) {
+    UNUSED(p, item);
+    return ((RdbxFilter *) userData)->cbReturnValue;
+}
+
+static RdbRes filterSetIS(RdbParser *p, void *userData, RdbBulk intset) {
+    UNUSED(p, intset);
+    return ((RdbxFilter *) userData)->cbReturnValue;
+}
+
+static RdbRes filterSetLP(RdbParser *p, void *userData, RdbBulk listpack) {
+    UNUSED(p, listpack);
+    return ((RdbxFilter *) userData)->cbReturnValue;
+}
+
+static RdbRes filterFunction(RdbParser *p, void *userData, RdbBulk func) {
+    UNUSED(p, func);
+    return ((RdbxFilter *) userData)->cbReturnValue;
+}
+
+static RdbRes filterModule(RdbParser *p, void *userData, RdbBulk moduleName, size_t serializedSize) {
+    UNUSED(p, moduleName, serializedSize);
+    return ((RdbxFilter *) userData)->cbReturnValue;
+}
+
+/*** Handling raw ***/
+
+static RdbRes filterFrag(RdbParser *p, void *userData, RdbBulk frag) {
+    UNUSED(p, frag);
+    return ((RdbxFilter *) userData)->cbReturnValue;
+}
+
+static RdbRes filterRawBegin(RdbParser *p, void *userData, size_t size) {
+    UNUSED(p, size);
+    return ((RdbxFilter *) userData)->cbReturnValue;
+}
+
+static RdbRes filterRawEnd(RdbParser *p, void *userData) {
+    UNUSED(p);
+    return ((RdbxFilter *) userData)->cbReturnValue;
+}
+
+/*** common init ***/
+
+static void defaultFilterDataCb(RdbHandlersDataCallbacks *dataCb) {
+    memset(dataCb, 0, sizeof(*dataCb));
+    dataCb->handleNewKey = filterNewKey;
+    dataCb->handleEndKey = filterEndKey;
+    dataCb->handleNewDb = filterNewDb;
+    dataCb->handleDbSize = filterDbSize;
+
+    dataCb->handleStringValue = filterString;
+    dataCb->handleListItem = filterList;
+    dataCb->handleHashField = filterHash;
+    dataCb->handleSetMember = filterSetMember;
+    dataCb->handleFunction = filterFunction;
+    dataCb->handleModule = filterModule;
+}
+
+static void defaultFilterStructCb(RdbHandlersStructCallbacks *structCb) {
+    memset(structCb, 0, sizeof(*structCb));
+    /* common */
+    structCb->handleNewKey = filterNewKey;
+    structCb->handleEndKey = filterEndKey;
+    structCb->handleNewDb = filterNewDb;
+    structCb->handleDbSize = filterDbSize;
+
+    /* string */
+    structCb->handleString = filterString;
+    /* list */
+    structCb->handleListLP = filterListLP;
+    structCb->handleListZL = filterListZL;
+    structCb->handleListPlain = filterListPlain;
+    /* hash */
+    structCb->handleHashPlain = filterHashPlain;
+    structCb->handleHashZL = filterHashZL;
+    structCb->handleHashLP = filterHashLP;
+    structCb->handleHashZM = filterHashZM;
```

Comment:
The new callback will be called.

Reply:
i mean if in the future we'll add `handleHashZZ`, and forget to add it here in the filter.
everything will appear to work correctly except for that one rdb format that we're unlikely to test.
since these are all boilerplate callbacks, and are likely to be forgotten, i think that we can (and should) do something so that it won't happen.

Reply:
I don't think it is worth to put an effort into it and additonal logic. If someone tomorrow will wants to add a new callback, he should lookup on similar callback that already supported and mimic all the relevant places.